app/models/katello/product.rb

```diff
@@ -31,8 +31,11 @@ class Product < Katello::Model
     has_many :alternate_content_source_products, dependent: :delete_all, inverse_of: :product, :class_name => "Katello::AlternateContentSourceProduct"
     has_many :alternate_content_sources, :through => :alternate_content_source_products, :inverse_of => :products, :class_name => "Katello::AlternateContentSource"
 
+    before_validation :set_provider_type_from_id, only: [:create]
```

Comment:
I think you already fixed the create path, or maybe it should go into some Dynflow action or similar?

Reply:
Yeah, I wasn't sure I needed both this and the validation, but it's nice to have. I'll see if anything breaks without it :P

Reply:
Removed the before_validation hook.